test/Makefile.am

```diff
@@ -4,6 +4,9 @@
 # Find include directory.
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
+# Need optimization disabled for correct behavior of some tests
+CFLAGS += -g -O0
```

Comment:
We must not set CFLAGS. There is an AM_CFLAGS you can use instead, if you must.

But instead, consider erroring out in configure if optimization behavior is not correct, instead of guessing flags.